File: shopthoitrang-server/src/models/banner.model.js
class Banner {
  constructor({ mabanner, duongdananh, mota, thutuhienthi, danghoatdong }) {
    this.mabanner = mabanner ?? null;
    this.duongdananh = duongdananh;
    this.mota = mota ?? null;
    this.thutuhienthi = thutuhienthi ?? null;
    this.danghoatdong = typeof danghoatdong === 'boolean' ? danghoatdong : true;
  }

  toJSON() {
    return {
      mabanner: this.mabanner,
      duongdananh: this.duongdananh,
      mota: this.mota,
      thutuhienthi: this.thutuhienthi,
      danghoatdong: this.danghoatdong,
    };
  }
}
module.exports = Banner;

File: shopthoitrang-mobile/lib/models/banner_model.dart
class BannerModel {
  final int? id; // mabanner
  final String imageUrl; // duongdananh
  final String? description; // mota
  final int? order; // thutuhienthi
  final bool active; // danghoatdong

  BannerModel({
    this.id,
    required this.imageUrl,
    this.description,
    this.order,
    required this.active,
  });

  factory BannerModel.fromJson(Map<String, dynamic> json) {
    return BannerModel(
      id: json['mabanner'] ?? json['maBanner'] ?? json['id'],
      imageUrl: (json['duongdananh'] ?? json['duongDanAnh'] ?? '').toString(),
      description: json['mota'] ?? json['moTa'],
      order: json['thutuhienthi'] ?? json['thuTuHienThi'],
      active: json['danghoatdong'] is bool
          ? json['danghoatdong']
          : (json['dangHoatDong'] is bool ? json['dangHoatDong'] : true),
    );
  }
}
